src/pudl/helpers.py

```diff
@@ -2263,3 +2265,59 @@ def retry(
             )
             time.sleep(delay)
     return func(**kwargs)
+
+
+def get_parquet_table(
+    table_name: str,
+    columns: list[str] | None = None,
+    filters: list[tuple[str, str, Any]]
+    | list[list[tuple[str, str, Any]]]
+    | None = None,
+) -> pd.DataFrame:
+    """Read a table from Parquet files with optional column selection and filtering.
+
+    This function provides a general-purpose interface for reading PUDL tables from
+    Parquet files. It supports selective column reading for performance, optional
+    filters for data subsetting, and automatic schema validation.
+
+    Args:
+        table_name: Name of the table to read.
+        columns: List of columns to read. If None, all columns are read.
+        filters: Optional filters to apply when reading the Parquet file. See the
+            :func:`pyarrow.parquet.read_table` documentation for details on filter
+            syntax. If None, no filters are applied.
+
+    Returns:
+        DataFrame with the requested data, with PUDL schema validation applied.
+
+    Raises:
+        FileNotFoundError: If the Parquet file for the table doesn't exist.
+        ValueError: If the table_name is not a valid PUDL resource.
+    """
+    # Import here to avoid circular imports
+    from pudl.metadata.classes import Resource
+
+    paths = PudlPaths()
+
+    # Get the Parquet file path
+    parquet_path = paths.parquet_path(table_name)
+
+    # Get the schema for validation
+    resource = Resource.from_id(table_name)
+    pyarrow_schema = resource.to_pyarrow()
+
+    # Read the Parquet file
+    df = pq.read_table(
+        source=parquet_path,
+        schema=pyarrow_schema,
+        columns=columns,
+        filters=filters,
+        use_threads=True,
+        memory_map=True,
+    ).to_pandas()
+
+    # Only enforce schema if we're reading all columns
+    if columns is None:
+        return resource.enforce_schema(df)
+    # For specific columns, apply PUDL dtypes for the columns we have
+    return apply_pudl_dtypes(df, group=resource.field_namespace)
```

Comment:
The need for this functionality was showing up in a couple of places, so I consolidated it. PudlTabl and the PudlParquetIOManager both use this now, as do the tests that need to load data directly.